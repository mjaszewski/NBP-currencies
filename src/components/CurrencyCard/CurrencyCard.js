import React, {Component} from 'react';
import PropTypes from "prop-types";PropTypes

import {Card, Button} from "semantic-ui-react";

class CurrencyCard extends Component {

    componentDidMount() {
    }

    handleDelete = (id) => {
        this.props.handleDelete(id)
    };

    render() {
        const { name, id, value} = this.props;
        return (
            <Card>
                <Card.Content>
                    <Card.Header>{name}</Card.Header>
                    <Card.Meta>{id}</Card.Meta>
                    <Card.Description>
                        <strong>{value}</strong>
                    </Card.Description>
                </Card.Content>
                <Card.Content extra>
                    <div className='ui one buttons'>
                        <Button basic negative onClick={this.handleDelete(id)}>
                            Remove
                        </Button>
                    </div>
                </Card.Content>
            </Card>
        );
    }
}

CurrencyCard.propTypes = {
    name: PropTypes.string,
    id: PropTypes.string,
    value: PropTypes.number,
    handleDelete: PropTypes.func

};

export default CurrencyCard