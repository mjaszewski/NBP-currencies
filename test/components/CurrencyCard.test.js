import CurrencyCard from '../../src/components/CurrencyCard/CurrencyCard';
import {Button, Card} from "semantic-ui-react";

const handleDelete = jest.fn();

function mountSetup() {
    const props = {
        name: "bat (Tajlandia)",
        code: "THB",
        value: 0.1252,
        date: "2020-02-21",
        handleDelete: handleDelete,
    };

    const wrapper = mount(<CurrencyCard {...props} />);

    return {
        props,
        wrapper
    };
}

let wrapper;
let props;

beforeEach(() => {
    const setup = mountSetup();
    wrapper = setup.wrapper;
    props = setup.props;
});

describe('Render Currency Card', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should render a card with the details', () => {
        expect(wrapper.find(Button)).toHaveLength(1);
        expect(wrapper.find(Card.Header).text()).toBe(`${props.name} - ${props.date}`);
        expect(wrapper.find(Card.Meta).text()).toBe(props.code);
        expect(wrapper.find(Card.Description).text()).toBe(`${String(props.value)} zÅ‚`);
    });
});

describe('Currency Card interactions', () => {
    it('should call the onClick function when \'Remove\' button is clicked', () => {
        wrapper.find(Button).simulate('click');
        expect(handleDelete).toHaveBeenCalledTimes(1);
    });
});
