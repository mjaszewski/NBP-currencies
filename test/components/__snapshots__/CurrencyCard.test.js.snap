// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Currency Card renders correctly 1`] = `
<CurrencyCard
  code="THB"
  date="2020-02-21"
  handleDelete={
    [MockFunction] {
      "calls": Array [
        Array [
          "THB",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  name="bat (Tajlandia)"
  value={0.1252}
>
  <Card>
    <div
      className="ui card"
      onClick={[Function]}
    >
      <CardContent>
        <div
          className="content"
        >
          <CardHeader>
            <div
              className="header"
            >
              bat (Tajlandia)
               - 
              2020-02-21
            </div>
          </CardHeader>
          <CardMeta>
            <div
              className="meta"
            >
              THB
            </div>
          </CardMeta>
          <CardDescription>
            <div
              className="description"
            >
              <strong>
                0.1252
                 z≈Ç
              </strong>
            </div>
          </CardDescription>
        </div>
      </CardContent>
      <CardContent
        extra={true}
      >
        <div
          className="extra content"
        >
          <div
            className="ui one buttons"
          >
            <Button
              as="button"
              basic={true}
              className="remove-favourite"
              negative={true}
            >
              <Ref
                innerRef={
                  Object {
                    "current": <button
                      class="ui basic negative button remove-favourite"
                    >
                      Remove
                    </button>,
                  }
                }
              >
                <RefFindNode
                  innerRef={
                    Object {
                      "current": <button
                        class="ui basic negative button remove-favourite"
                      >
                        Remove
                      </button>,
                    }
                  }
                >
                  <button
                    className="ui basic negative button remove-favourite"
                    onClick={[Function]}
                  >
                    Remove
                  </button>
                </RefFindNode>
              </Ref>
            </Button>
          </div>
        </div>
      </CardContent>
    </div>
  </Card>
</CurrencyCard>
`;
